servicetalk-concurrent-api/src/test/java/io/servicetalk/concurrent/api/publisher/TimeoutPublisherTest.java

```diff
@@ -166,28 +210,55 @@ public void noDataAndTimeout() {
         assertThat(testExecutor.scheduledTasksExecuted(), is(1));
     }
 
-    @Test
-    public void dataAndTimeout() {
-        init();
+    @ParameterizedTest(name = "{displayName} [{index}] {arguments}")
+    @EnumSource(TimerBehaviorParam.class)
+    public void dataAndTimeout(TimerBehaviorParam params) throws Exception {
+        init(params, Duration.ofSeconds(2));
 
+        assertThat(testExecutor.scheduledTasksPending(), is(1));
+        assertThat(testExecutor.scheduledTasksExecuted(), is(0));
         subscriber.awaitSubscription().request(10);
-        assertThat(subscriber.pollOnNext(10, MILLISECONDS), is(nullValue()));
-        assertThat(subscriber.pollTerminal(10, MILLISECONDS), is(nullValue()));
-        publisher.onNext(1, 2, 3);
-        assertThat(subscriber.takeOnNext(3), contains(1, 2, 3));
+        Supplier<Throwable> timedOut = null;
+        for (int x = 1; x <= 3; x++) {
+            assertThat(subscriber.pollOnNext(0, MILLISECONDS), is(nullValue()));
+            timedOut = subscriber.pollTerminal(0, MILLISECONDS);
+            if (null != timedOut) {
+                break;
+            }
+            publisher.onNext(x); // may reset timer
+            SECONDS.sleep(1);
```

Comment:
can you reduce the time for this test. takes ~7 seconds for both permutations to run on my local host.

Reply:
I was worried that it would be too long. I will try 100ms "steps" rather than 1s.